### PR TITLE
Modify logic to calculate snapping after animation

### LINKED ISSUE #146: how to make swipe threshold smaller
I have 3 snapPoints, the highest one being the screen height, this causes a problem however, when you try to swipe up, you really have to drag your finger really high in the screen to make the bottom sheet snap, I tried changing the value of `callbackThreshold` but it doesn't seem to affect how far you need to drag the sheet.

Any suggestions on how to solve this?

### PR DESCRIPTION
**Pull request by ospfranco**
Right now if the modal takes all the screen size the gesture needed to expand it is really unnatural and feels unresponsive, I've changed the logic to make it feel snappier.

Fixes #146

### CODE REVIEW

**Comment by mich4xD** [comment]
It seems that this pull request breaks the logic for the initial snap position in case the initialSnap is set to 0 (hidden) for the following snap points [0,750].
The use case I need this for is hiding the bottom drawer until some one clicks on some view and then I expand it.

Code it can be tested with:

      ```
       <View
            style={{ height: "100%", width: "100%", backgroundColor: "red" }}
            onTouchStart={() =>
              this._rideInformationDrawerRef.current.snapTo(1)
            }
          >
            <Text>{"Press to show drawer"}</Text>
          </View>
          <BottomSheet
            ref={this._rideInformationDrawerRef}
            initialSnap={0}
            snapPoints={[0, 750]}
            renderContent={() => (
              <View style={{height: '100%', width:'100%', backgroundColor:'blue'/>
            )}
          />
```

Let me know if you've got any further questions about this.

**Comment by ospfranco** [comment]
I believe the snap points need to be in a reverse order, did you try setting them as [750, 0]?

**Comment by mich4xD** [comment]
@ospfranco, unfortunently having the initial snpapoint as 0 for the bottomsheet to be hidden initially still does not work even when flipping the order of the array to the proper one.
I

Code: 

`this._drawerRef= React.createRef();

       <View
            style={{ height: "100%", width: "100%", backgroundColor: "red" }}
            onTouchStart={() =>
              this._drawerRef.current.snapTo(1)
            }
        >
            <Text>{"Press to show drawer"}</Text>
          </View>
          <BottomSheet
            ref={this._drawerRef}
            initialSnap={1}
            snapPoints={[750, 0]}
            renderContent={() => (
              <View style={{height: '100%', width:'100%', backgroundColor:'blue'/>
            )}
          />`

**Comment by ospfranco** [comment]
can you try now? it seems to be fixed

**Comment by mich4xD** [comment]
@ospfranco, It's working properly now. Thanks!

File: src/index.tsx [hunk 1]
```diff
@@ -339,10 +339,19 @@ export default class BottomSheetBehavior extends React.Component<Props, State> {
...
-    const middlesOfSnapPoints: Animated.Node<number>[] = []
+    const middlesOfSnapPoints: [
+      Animated.Node<number>,
+      Animated.Node<number>
+    ][] = []
+
     for (let i = 1; i < snapPoints.length; i++) {
-      middlesOfSnapPoints.push(divide(add(snapPoints[i - 1], snapPoints[i]), 2))
+      const tuple: [Animated.Node<number>, Animated.Node<number>] = [
+        add(snapPoints[i - 1], 10),
```
Comment (zmnv) [inline src/index.tsx]: Recommend to add `sub(snapPoints[i], 25),` after `add`:

```
add(snapPoints[i - 1], 10),
sub(snapPoints[i], 25),
```

**Comment by lucbas** [comment]
Why has this PR been closed?

I think it is highly relevant and needed.

**Comment by ospfranco** [comment]
Is this repo active? The PR has been opened for a long time and I got tired of receiving notifications, I get the confirmation that it will get merged I'll re open it, otherwise you guys can open a new one

**Comment by lucbas** [comment]
Yes, the repo is definitely still active with that many people using it. @osdnk has not been that active recently, but I'm pretty sure that did not abandon this repo.

Also, other folks such as @BrtqKr and @davidgovea have been helping and maybe it would make sense to look for other maintainers too.
We rely heavily on this repo too and are willing to help out in areas where help is wanted. For example, the things that bother us the most is this exact snap point logic and virtualization. But that discussion is not part of this PR.

**Comment by osdnk** [comment]
Sorry for a long waiting. I think it’s ok to merge it

**Comment by ospfranco** [comment]
Change suggested by @zmnv is implemented, you can merge it if you like

**Pull request merged**